File: frontend/src/types/registration.ts
export type RegistrationItem = {
  id: string;
  user_id: string;
  event_id: string;
  status: "registered" | "cancelled" | "waitlist";
  registered_at: string;
  notes: string | null;
};

export type MyRegistrationsResponse = {
  items: RegistrationItem[];
  total: number;
};

export type EventRegistrationUserItem = {
  user_id: string;
  full_name: string;
  email: string;
  status: "registered" | "cancelled" | "waitlist";
  registered_at: string;
};

export type EventRegistrationsResponse = {
  items: EventRegistrationUserItem[];
  total: number;
};

File: backend/app/schemas/registration.py
from datetime import datetime

from pydantic import BaseModel, Field


class RegistrationCreate(BaseModel):
    notes: str | None = Field(default=None, max_length=1000)


class RegistrationResponse(BaseModel):
    id: str
    user_id: str
    event_id: str
    status: str
    registered_at: datetime
    notes: str | None


class MyRegistrationsResponse(BaseModel):
    items: list[RegistrationResponse]
    total: int
